README.md

```diff
@@ -28,13 +27,298 @@ pip install -U pyinstaller-hooks-contrib
+Good sources of such minimal tests are the hello world examples
+from the documentation of whichever library you're writing a hook for.
+Tests should not explicitly check for presence of data files or hidden modules directly -
+rather they should use parts of the library which are expected to use said data files or hidden modules.
+
+Tests currently all live in [src/_pyinstaller_hooks_contrib/tests/test_libraries.py](../master/src/_pyinstaller_hooks_contrib/tests/test_libraries.py).
+Navigate there and add something like the following, replacing all occurrences of `foo` with the real name of the library.
+(Note where you put it in that file doesn't matter.)
+
+```python
+@importorskip('foo')
+def test_foo(pyi_builder):
+    pyi_builder.test_source("""
+
+	# Your test here!
+        import foo
+
+	foo.something_fooey()
+
+    """)
+```
+
+If need be, you can specify version constraints using `@importorskip('foo', minversion="1.2.3")`
+if the library has changed significantly between versions.
+
+
+#### Run the test locally
+
+Running our full test suite is not recommended as it will spend a very long time testing code which you have not touched.
+Instead, run tests individually using either the `-k` option to search for test names:
+
+```
+pytest -k test_foo
+```
+
+Or using full paths:
+```
+pytest src/_pyinstaller_hooks_contrib/tests/test_libraries.py::test_foo
+```
+
+
+#### Run the test on CI/CD
+
+To test hooks on all platforms we use Github's continuous integration (CI/CD).
+Our CI/CD is a bit unusual in that it's triggered manually and takes arguments
+which limit which tests are run.
+This is for the same reason we filter tests when running locally -
+the full test suite runs like a dead snail on ketamine.
+
+First push the changes you've made so far.
+
+```commandline
+git push --set-upstream origin hook-for-foo
+```
+
+Replace *billy-the-buffalo* with your Github username in the following url then open it.
+It should take you to the `oneshot-test` actions workflow on your fork.
+You may be asked if you want to enable actions on your fork - say yes.
+```
+https://github.com/billy-the-buffalo/pyinstaller-hooks-contrib/actions/workflows/oneshot-test.yml
+```
+
+Find the **Run workflow** button and click on it.
+A dialog should appear containing one drop-down menu and 5 line-edit fields.
+This dialog is where you specify what to test and which platforms and Python versions to test on.
+Its fields are as follows:
+1.  A branch to run from. Set this to the branch which you are using (e.g. ``hook-for-foo``),
+2.  Which package(s) to install and their version(s).
+    Which packages to test are inferred from which packages are installed.
+    * Set to `foo` to test the latest version of `foo`,
+    * Set to `foo==1.2, foo==2.3` (note the comma) to test two different versions of `foo` in separate jobs,
+    * Set to `foo bar` (note the lack of a comma) to test `foo` and `bar` in the same job,
```

Comment:
Is this right? I did that and got `ERROR: Invalid requirement: 'spacy srsly thinc'`: https://github.com/colin-fsa/pyinstaller-hooks-contrib/runs/2867341158?check_suite_focus=true

Reply:
Looks like you wrapped used `"spacy srsly thinc"` instead of just `spacy srsly thinc`?

Reply:
That seemed like the obvious thing to do since the default for that input is `"numpy==1.19", "numpy<1.18"`

Reply:
Yeah I see where you're coming from. It does appear to imply that each job should be quoted. The quotes are only needed to stop bash from interpreting `<` as a redirection. Other than that, they should never be needed. I'm thinking perhaps I could have it escape those characters itself so that there are no quotes involved at all...